source/Sashimi.Terraform.Tests/TerraformValidatorFixture.cs

```diff
@@ -0,0 +1,251 @@
+                {
+                    {TerraformSpecialVariables.Action.Terraform.AzureAccount, Boolean.TrueString},
+                }, new List<PackageReference>()));
+            result.IsValid.Should().BeFalse();
+            result.Errors.Single().PropertyName.Should().Be("Octopus.Action.AzureAccount.Variable");
+        }
+        
+        [Test]
+        public void Should_have_no_error_when_azure_account_is_specified()
+        {
+            var result = validator.TestValidate(new DeploymentActionValidationContext(TerraformActionTypes.Plan,
+                new Dictionary<string, string>
+                {
+                    {TerraformSpecialVariables.Action.Terraform.AzureAccount, Boolean.TrueString},
+                    {"Octopus.Action.AzureAccount.Variable", "Myvariable"},
+                }, new List<PackageReference>()));
+            result.IsValid.Should().BeTrue();
+        }
+        
+        [Test]
+        public void Should_have_error_when_aws_account_has_no_roles_specified()
+        {
+            var result = validator.TestValidate(new DeploymentActionValidationContext(TerraformActionTypes.Plan,
+                new Dictionary<string, string>
+                {
+                    {TerraformSpecialVariables.Action.Terraform.ManagedAccount, TerraformSpecialVariables.AwsAccount},
+                    {TerraformSpecialVariables.Action.Aws.AssumeRole, Boolean.TrueString},
+                    {TerraformSpecialVariables.Action.Aws.AwsRegion, "MyRegion"},
+                    {TerraformSpecialVariables.Action.Aws.AccountVariable, "MyVariable"},}, new List<PackageReference>()));
+            result.IsValid.Should().BeFalse();
+            result.Errors.Select(_ => _.PropertyName).Should().Equal(TerraformSpecialVariables.Action.Aws.AssumedRoleArn, TerraformSpecialVariables.Action.Aws.AssumedRoleSession);
+        }
+        
+        [Test]
+        public void Should_have_no_error_when_aws_account_has_roles_specified()
+        {
+            var result = validator.TestValidate(new DeploymentActionValidationContext(TerraformActionTypes.Plan,
+                new Dictionary<string, string>
+                {
+                    {TerraformSpecialVariables.Action.Terraform.ManagedAccount, TerraformSpecialVariables.AwsAccount},
+                    {TerraformSpecialVariables.Action.Aws.AssumeRole, Boolean.TrueString},
+                    {TerraformSpecialVariables.Action.Aws.AwsRegion, "MyRegion"},
+                    {TerraformSpecialVariables.Action.Aws.AccountVariable, "MyVariable"},
+                    {TerraformSpecialVariables.Action.Aws.AssumedRoleArn, "RoleArn"},
+                    {TerraformSpecialVariables.Action.Aws.AssumedRoleSession, "RoleSession"},
+                }, new List<PackageReference>()));
+            result.IsValid.Should().BeTrue();
+        }
+        
+        [Test]
+        public void Should_have_error_when_aws_account_with_no_region_specified()
```

Comment:
A newline between `Act` and `Assert` would make this more readable.

Also consider pulling out a variable for the context, this way the difference in setup is more prominent:
![image](https://user-images.githubusercontent.com/2888279/82162654-871f5180-98e9-11ea-90ae-309226bad750.png)

Reply:
fixed